### ISSUE TITLE
Search Dialog Box irritating behavior

### ISSUE THREAD

**Issue by KPreston2900050** [notepad-plus-plus/notepad-plus-plus#12795]
I use two monitors.  I have my main working monitor which is also used for N++
The second one is for watching video's or Programing documentation (PDF) and what not.
A lot of the time my second monitor is off.

The issue is with every new instance of N++ and I use Ctrl-F search the dialog box defaults to the second monitor.
This forces me to turn it on if it's not on and drag the dialog box to my main monitor which is default for all programs on my PC.

Windows 11/x64 dual monitors.

This has been going on for as long as I started using more than one monitor and I am currently using v8.4.8

As a side note, I stopped using install because it always overwrote my preferences.
Portable allows me to save my configurations.
When I tried to fix this Dialog behavior I went ahead and started fresh which did not fix the problem and spent a good hour reconfiguration everything back the way I need it.

This is not a complaint, I understand it is free and I use this more than any other editor for programing Z80 assembly.
However, if there is a way to preserve your previous settings that would be cool.

**Comment by alankilborn** [comment]
>A lot of the time my second monitor is off.

It boggles the mind that if you have a second monitor why it ever gets turned off.
I have 5 monitors (all on) and sometimes that isn't even enough.

>every new instance

What does this mean, exactly?  Are you launching new instances from a main N++ instance (main instance stays running)?  Or do you mean, single N++ running always and your use of "instance" means basically a run-quit cycle?

_Debug Info_ would have been good to see, instead of the request in the issue template being ignored.

**Comment by alankilborn** [comment]
Not sure at all why you brought up Windows Explorer...

You really didn't clear up the "instance" thing I asked about...  Nor did you supply _Debug Info_...

I was trying to get to the root cause of why this is happening for you, but we may never get there...

Notepad++ 8.4.8, for me, remembers the position where I last put the _Find_ window, and restores it to that position when I reinvoke Find.  I suppose if the first time this happened, it was "offscreen", I'd somehow get it back on screen...but from then on, it recalls itself to the position I set most recently.

It isn't a fix because I don't yet (and probably won't) know what the real situation is, but if you press Ctrl+f and the _Find_ window doesn't seem to appear, press Ctrl+f again.  This second (in a row) Ctrl+f press tells the _Find_ window to recenter itself on the Notepad++ parent window.

**Comment by anishsane** [comment]
This issue bothers me as well.
My monitors are all on. But the find window opens on that monitor where it was opened last time.
Windows OS tracks the 2 windows (main window and find window) separately.

I hope npp developer team add some code logic to open the find window on the same monitor as the notepad++ main window.
Else, I have to search on all the monitors for where the find window is.

Moreover, it also confuses me (and probably others) at times, if the find window is open at all or not. Because the ctrl+f may not be received by npp because of some modal window.

**Comment by alankilborn** [comment]
>But the find window opens on that monitor where it was opened last time.

Yes, this is absolutely desired behavior.  Windows should stay where the user has put them.

>I hope npp developer team add some code logic to open the find window on the same monitor as the notepad++ main window.

No, this is not wanted.  Notepad++'s _Find_ window is so large that it is not reasonable to have it on the same monitor as the N++ main window, when N++ is in typical usage maximized -- the _Find_ window just covers too much of the main window.  On a single monitor system, users just have to suffer with this (or close the _Find_ window).

The answer to locating the window and putting it on the same monitor as N++ main window, is above; since apparently it was missed, I'll repeat it here:

_**"...if you press Ctrl+f and the Find window doesn't seem to appear, press Ctrl+f again. This second (in a row) Ctrl+f press tells the Find window to recenter itself on the Notepad++ parent window."**_

**Comment by anishsane** [comment]
@alankilborn , My issue is not with find window opening on the monitor where find window was moved last time.

My problem is: (e.g.)
If I move npp main window to monitor 1, I move the find window to monitor 1. (manually)
Next time, if I move the npp main window to monitor 2, the find window continues to open at monitor 1. And I have to move it manually.
This issue is especially relevant on laptops where we connect/disconnect the external monitor(s).

>No, this is not wanted.
Many other softwares (browsers, windows explorer, MSoffice softwares etc) have this behavior - a child window opened from the main window appears on the same screen as the main window. This may or may not be wanted, but from UX POV, this is what people are accustomed to.
Maybe you could add a boolean preference for this behavior, since the existing behavior and our ask look opinionated. Some may want this way, some the other way - and they can choose based on the boolean setting.

>> Notepad++'s Find window is so large 
I don't completely agree with this. Even with just 1960x1080 resolution, I am able to see it well enough. I tend to move it to right side of the main window.
But again, this is my opinion. YMMV.

Having said all this, I learned yesterday (on another thread) the double ctrl+f way to move the find window to the same monitor as main window. And I am now in process of changing my habit to press ctrl+f twice.

I wrote this comment just to convey, that while your points are valid, there are many users who find this behavior unconventional and hence (to quote this issue's title) irritating,

**Comment by alankilborn** [comment]
Windows need to stay where the user puts them.  Period, end of story.  Obviously I mean this in the sense of a closed window that is reopened later.  It doesn't matter where a "main" window is moved to, after the fact.

Obviously, this can be a problem if the location/position of restoration of a window no longer exists.  This is not always "well detectable" as monitor configurations can change, and while a location may "exist" it may not be the same (in user's view) as previously.  Notepad++ chooses to mitigate this with a second possible "Ctrl+f" press to recenter-on-parent.

--------------------

>changing my habit to press ctrl+f twice

I can't envision any possibly reasonable workflow where Ctrl+f has to be pressed twice every time one wants to recall the window.

-------------------

>Many other softwares (browsers, windows explorer, MSoffice softwares etc) have this behavior - a child window opened from the main window appears on the same screen as the main window. 

And this is what Notepad++ does as well.  ...Until the user repositions that window; from that moment on the child window reappears at the position of user placement.  If there's some "UX" guideline that indicates otherwise, please directly cite it.

If I run Notepad++ maximized on one monitor, and I invoke _Find_ with Ctrl+f and the window appears right in the middle of the same monitor (covering a create deal of the material I'm trying to work on), I grab the window title bar and move it to a second monitor.  I work with the _Find_ window and perhaps I leave it open or maybe I close it.  Say I close it.  When I need it for the next search I invoke Ctrl+f.  If the _Find_ window then appears centered on its Notepad++ parent, I would immediately remove Notepad++ from my system and start searching for a different text editor.

**Comment by anishsane** [comment]
If you open firefox settings or any window within it (clear history - ctrl+shift+del, e,g,) it will open the dialog on the same window as the window it was called from.
For that matter, if you have 2 FF windows open on different monitors and you press ctrl+shift+del, then the dialogue opens on the monitor that the active window is on.

>And this is what Notepad++ does as well. 

This is not how npp behaves. It opens at the place where find window was last moved.
EDIT: I was wrong about MS office behavior. MS office also behaves in the same way as npp.

> Until the user repositions that window; from that moment on the child window reappears at the position of user placement. 

This is not true either.
Use case:
Open npp from windows start menu. press ctrl+f. Search something.
Close the find window. Don't move the find window.
Move npp to different screen.
press ctrl+f again.
The find window continues to open on the old monitor.
I - the user -  have not positioned it on the old monitor.

>I can't envision any possibly reasonable workflow where Ctrl+f has to be pressed twice every time one wants to recall the window.

I keep switching npp window between my monitors as required. And as a user, I don't remember where the npp window was positioned last time.
And whenever I press ctrl+f, I first have to locate across all my monitors where my search window has opened. Hence the double ctrl+f.
This is especially relevant when I connect/disconnect the external monitor to the laptop. If I disconnect the monitor (even because of power cut), all windows are moved to laptop screen by OS, as expected.
After the power resumes or monitors are reconnected, I would move the main window back to the bigger monitor. And some time down the line, when I press ctrl+f, I would need to find the find window first, before I find a text with it. The find window would have opened on the laptop screen.

>Windows need to stay where the user puts them

Correct. For me, I put it on the same screen as my main window. If my main window moves, the find window does not move with it.
<ins>**Hence I suggested a boolean preference.**</ins>
* boolean option true => remember the absolute position of the find window across all the screens.  
* boolean option false => remember the position of the find window relative to the main window.  

Keep the default value of that boolean true, to keep the default behavior the same as now.  
But let the users decide how they want it to behave. Their expectations may be different from your expectations.

> Notepad++ chooses to mitigate this with a second possible "Ctrl+f" press to recenter-on-parent.  

And I thank the npp developers for this. I am changing my habit to use double ctrl+f for this.

And since it is "<ins>**_END OF STORY_**</ins>" for you, I will not be commenting on this thread further. I can live with the double ctrl+f as a workaround for the problem _we_ are facing.
